src/depfile_parser.cc

```diff
@@ -294,41 +294,32 @@ bool DepfileParser::Parse(string* content, string* err) {
     }
 
     if (len > 0) {
-      if (is_dependency) {
-        if (have_secondary_target_on_this_rule) {
-          if (!have_newline_since_primary_target) {
-            *err = "depfile has multiple output paths";
-            return false;
-          } else if (options_.depfile_distinct_target_lines_action_ ==
-                     kDepfileDistinctTargetLinesActionError) {
-            *err =
-                "depfile has multiple output paths (on separate lines)"
-                " [-w depfilemulti=err]";
+      StringPiece piece = StringPiece(filename, len);
+      // If we've seen this as an input before, skip it.
+      vector<StringPiece>::iterator pos = find(ins_.begin(), ins_.end(), piece);
```

Comment:
would a std::set make more sense here? All of the items in the vector need to be unique, right?

Reply:
`ins_` is the list of inputs and must preserve order, but one could consider using a separate local `std::set`.

Reply:
Actually since depfile-specified inputs do not affect `$in` the order does not matter.  One could convert the depfile parser's `ins_` member from `vector` to `std::set`.

Reply:
I think converting it over is a larger change out-of-scope here. Lots of test suite checks would need to cope with the out-of-order-ness of a `std::set`.

Reply:
Okay, that optimization can be reserved for a later PR.